### PR TITLE
Enable torch.nn.init._calculate_correct_fan in dynamo tracing

### LINKED ISSUE #102340: [Dynamo] Support several missing torch.* ops
### 🐛 Describe the bug

This is the list of torch.* ops that we are missing support, but was used frequently in 14k github models:
* torch.is_complex
* torch.get_autocast_gpu_dtype
* torch.nn.init._calculate_correct_fan

### Versions

N/A

cc @ezyang @msaroufim @wconstab @ngimel @bdhirsh @anijain2305 @voznesenskym @penguinwu @EikanWang @jgong5 @Guobing-Chen @XiaobingSuper @zhuhaozhe @blzheng @Xia-Weiwen @wenzhe-nrv @jiayisunx

### PR DESCRIPTION
**Pull request by fduwjj**
Stack from [ghstack](https://github.com/ezyang/ghstack) (oldest at bottom):
* __->__ #103182

cc @voznesenskym @penguinwu @anijain2305 @EikanWang @jgong5 @Guobing-Chen @XiaobingSuper @zhuhaozhe @blzheng @Xia-Weiwen @wenzhe-nrv @jiayisunx @ipiszy

### CODE REVIEW

File: torch/_dynamo/variables/torch.py [hunk 1]
```diff
@@ -496,6 +496,10 @@ def get_state_from_generator():
             return TorchVariable(torch.add, **options).call_function(
                 tx, [args[0], result], {}
             )
+        elif self.value == torch.nn.init._calculate_correct_fan:
+            return UserFunctionVariable(
+                torch.nn.init._calculate_correct_fan, **options
+            ).call_function(tx, args, {})
```
Comment (yanboliang) [inline torch/_dynamo/variables/torch.py:502]: This helped us to prove the idea that we can use inlining as the ultimate fallback for ```call_function```, we will do code refactor to remove redundant code later. Thanks!

**Review by yanboliang — APPROVED** [review]

**Comment by fduwjj** [comment]
Manually rebase the code.

**Comment by jansel** [comment]
I think this one could have been done by just adding the function to allowed_functions.py

**Comment by fduwjj** [comment]
@jansel I was trying to add this function to torch_object_ids inside `_allowed_function_ids`, but looks like the test fails. 

Below is the debug log:

```
----------------------------------------------------------- Captured stdout call ------------------------------------------------------------
unimplemented [('torch.* op returned non-Tensor int call_function <function _calculate_correct_fan at 0x7f3c2a6a5310>', 1)]
----------------------------------------------------------- Captured stderr call ------------------------------------------------------------
[2023-06-08 20:39:38,419] torch._dynamo.eval_frame: [DEBUG] skipping __init__ /fsx/users/fduwjj/conda/envs/pytorch/lib/python3.9/contextlib.py
[2023-06-08 20:39:38,420] torch._dynamo.eval_frame: [DEBUG] skipping __enter__ /fsx/users/fduwjj/conda/envs/pytorch/lib/python3.9/contextlib.py
[2023-06-08 20:39:38,420] torch._dynamo.eval_frame: [DEBUG] skipping __init__ /fsx/users/fduwjj/conda/envs/pytorch/lib/python3.9/contextlib.py
[2023-06-08 20:39:38,420] torch._dynamo.eval_frame: [DEBUG] skipping __enter__ /fsx/users/fduwjj/conda/envs/pytorch/lib/python3.9/contextlib.py
[2023-06-08 20:39:38,420] torch._dynamo.eval_frame: [DEBUG] skipping enable_dynamic /fsx/users/fduwjj/work/pytorch/torch/_dynamo/eval_frame.py
[2023-06-08 20:39:38,420] torch._dynamo.symbolic_convert: [INFO] Step 1: torchdynamo start tracing test_get_calculate_correct_fan /fsx/users/fduwjj/work/pytorch/test/dynamo/test_functions.py:444
[2023-06-08 20:39:38,421] torch._dynamo.variables.builder: [DEBUG] wrap_to_fake L['x'] (10, 10) None None
[2023-06-08 20:39:38,422] torch._dynamo.symbolic_convert: [DEBUG] TRACE starts_line /fsx/users/fduwjj/work/pytorch/test/dynamo/test_functions.py:446
[2023-06-08 20:39:38,422] torch._dynamo.symbolic_convert: [DEBUG] TRACE LOAD_GLOBAL torch []
[2023-06-08 20:39:38,422] torch._dynamo.symbolic_convert: [DEBUG] TRACE LOAD_ATTR nn [TorchVariable(<module 'torch' from '/fsx/users/fduwjj/work/pytorch/torch/__init__.py'>)]
[2023-06-08 20:39:38,426] torch._dynamo.symbolic_convert: [DEBUG] TRACE LOAD_ATTR init [TorchVariable(<module 'torch.nn' from '/fsx/users/fduwjj/work/pytorch/torch/nn/__init__.py'>)]
[2023-06-08 20:39:38,426] torch._dynamo.symbolic_convert: [DEBUG] TRACE LOAD_ATTR _calculate_correct_fan [TorchVariable(<module 'torch.nn.init' from '/fsx/users/fduwjj/work/pytorch/torch/nn/init.py'>)]
[2023-06-08 20:39:38,427] torch._dynamo.symbolic_convert: [DEBUG] TRACE LOAD_FAST x [TorchVariable(<function _calculate_correct_fan at 0x7f3c2a6a5310>)]
[2023-06-08 20:39:38,427] torch._dynamo.symbolic_convert: [DEBUG] TRACE LOAD_CONST fan_in [TorchVariable(<function _calculate_correct_fan at 0x7f3c2a6a5310>), TensorVariable()]
[2023-06-08 20:39:38,427] torch._dynamo.symbolic_convert: [DEBUG] TRACE CALL_FUNCTION 2 [TorchVariable(<function _calculate_correct_fan at 0x7f3c2a6a5310>), TensorVariable(), ConstantVariable(str)]
```

Am I missing something here?

**Comment by jansel** [comment]
Ah sorry.  I think you want it inlined not put in the graph.  Inlining is controlled from skipfiles.

This change is fine, we may want to add a list of force-inline functions at some point in the future.

**Pull request closed without merging**